### ISSUE TITLE
can't source $nu.config-path from keybinding

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#5794]
### Describe the bug

Hit F5 and I get this error now. Previously it worked.
```
Error:
  × error within history: No command run

error: process didn't exit successfully: `C:\CarTar\debug\nu.exe` (exit code: 1)
```
/cc @sholderbach - this sounds like an error you reported with `No command run`

### How to reproduce

have a keybinding like this
```
    {
        name: reload_config
        modifier: none
        keycode: f5
        mode: emacs
        event: {
        send: executehostcommand,
        cmd: $"source '($nu.config-path)'"
        }
    }
```

### Expected behavior

source the config

### Configuration

|key|value|
|-|-|
|version|0.64.0|
|branch|main|
|commit_hash|cb66d2bcad6db50a71d0accba0f76afe0ec12cac|
|build_os|windows-x86_64|
|build_target|x86_64-pc-windows-msvc|
|rust_version|rustc 1.61.0 (fe5b13d68 2022-05-18)|
|rust_channel|stable-x86_64-pc-windows-msvc|
|cargo_version|cargo 1.61.0 (a028ae42f 2022-04-29)|
|pkg_version|0.64.0|
|build_time|2022-06-15 11:04:37 -05:00|
|build_rust_channel|debug|
|features|database, dataframe, default, trash, which, zip|
|installed_plugins|gstat|

**Comment by elferherrera** [comment]
I don't think this will work because it needs to evaluate the value of $nu.config-path. if you replace it with the abs path it should work

**Comment by fdncred** [comment]
> I don't think this will work because it needs to evaluate the value of $nu.config-path. if you replace it with the abs path it should work

I know that's the party line but guess what, it's worked since you implemented menus. It's always worked! Go try on 0.63 and see. :)

**Comment by fdncred** [comment]
> Weird that this triggers the history metadata extension

ya, i really don't understand that. Maybe executehostcommand does something weird?

**Comment by kubouch** [comment]
It works for me (tested on latest main). Here's my keybinding:
```
       {
           name: reload_config
           modifier: control
           keycode: char_r
           mode: [ emacs vi_insert vi_normal ]
           event: [
               { edit: clear }
               {
                  edit: insertString
                  value: $"source ($nu.env-path); source ($nu.config-path)"
               }
               { send: Enter }
           ]
        }
```

**Comment by sholderbach** [comment]
Ah did you change the history type in the config and reloaded the config directly without restarting the session... the repl instantiates the history once but queries the config for whether it has the metadata that now changed.

**Comment by fdncred** [comment]
> It works for me (tested on latest main). Here's my keybinding:

yours works for me too now. thanks @kubouch 

> Ah did you change the history type in the config and reloaded the config directly without restarting the session... the repl instantiates the history once but queries the config for whether it has the metadata that now changed.

maybe? i'm not sure. i'm in the process of switching all my environments over to the sqlite db so i could've been mid switch?

**Comment by fdncred** [comment]
oh, wait. I just noticed that @kubouch you're not using `executehostcommand`. i wonder if that has something to do with mine puking. let me re-write it and see.

**Comment by fdncred** [comment]
i guess not - maybe it's what stefan was saying?  This works now too and it doesn't print the mess on the command line.
```
    {
        name: reload_config
        modifier: none
        keycode: f5
        mode: [ emacs vi_insert vi_normal ]
        event: [
            { edit: clear }
            { send: executehostcommand cmd: $"source ($nu.env-path); source ($nu.config-path)" }
        ]
    }
```

**Comment by fdncred** [comment]
the above executehostcommand is working now

**Comment by fdncred** [comment]
@elferherrera figured out that if you hit F5 as the very first command, it pukes. If you do it after `ls` or anything else, it works. I suspect something isn't getting initialized right.

**Comment by fdncred** [comment]
@sholderbach I think I see what's going on here. This is the offending code in reedline.
```rust
    /// update the last history item with more information
    pub fn update_last_command_context(
        &mut self,
        f: &dyn Fn(HistoryItem) -> HistoryItem,
    ) -> crate::Result<()> {
        if let Some(r) = &self.history_last_run_id {
            self.history.update(*r, f)?;
        } else {
            return Err(ReedlineError(ReedlineErrorVariants::OtherHistoryError(
                "No command run",
            )));
        }
        Ok(())
    }
```
I think what's happening is that `self.history_last_run_id` is set to `None` and therefore if I hit F5 to reload the config, before any command is ever run, it's failing with the `No command run` error.

I'm wondering if, on launch, reedline should get the last history id from history.txt/.sqlite3 and populate `history_last_run_id` with that value? Thoughts?

**Comment by fdncred** [comment]
no longer an issue. i'm using this now and it works fine.
```
  {
    name: reload_config
    modifier: none
    keycode: f5
    mode: [ emacs vi_insert vi_normal ]
    event: [
      { edit: clear }
      { send: executehostcommand cmd: $"source ($nu.env-path); source ($nu.config-path)" }
    ]
  }
```
